### PR TITLE
fix failing lint : `findByPath` requires a specific ordering of project evaluation in `flutter.groovy`

### PR DESCRIPTION
**Pull request by AbdeMohlbi**
fix failing lint : `'findByPath' requires a specific ordering of project evaluation`
Capture d’écran 2024-11-21 230157
Capture d’écran 2024-11-21 230245

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter_tools/gradle/src/main/groovy/flutter.groovy [hunk 1]
```diff
@@ -1235,8 +1235,18 @@ class FlutterPlugin implements Plugin<Project> {
...
             // In add to app scenarios, a Gradle project contains a `:flutter` and `:app` project.
             // `:flutter` is used as a subproject when these tasks exists and the build isn't building an AAR.
-            Task packageAssets = project.tasks.findByPath(":flutter:package${variant.name.capitalize()}Assets")
-            Task cleanPackageAssets = project.tasks.findByPath(":flutter:cleanPackage${variant.name.capitalize()}Assets")
+            Task packageAssets
+            Task cleanPackageAssets
+            try {
+                packageAssets = project.tasks.named("package${variant.name.capitalize()}Assets").get()
+            } catch (UnknownTaskException ignored) {
+                packageAssets = null
```
Comment (AbdeMohlbi) [inline packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:1243]: ``findByPath` does return null if the a task with the specified name does not exist so it's basically the same behavior

**Review by reidbaker — APPROVED** [review]

**Comment by reidbaker** [comment]
test-exempt: refactor with no functional change.

**Comment by jesswrd** [comment]
@AbdeMohlbi Hey this PR looks really similar to this PR: https://github.com/flutter/flutter/pull/159301. I'm just wondering if there's a reason these PRs need to be split (like some sort of dependency)?

**Comment by AbdeMohlbi** [comment]
> @AbdeMohlbi Hey this PR looks really similar to this PR: #159301. I'm just wondering if there's a reason these PRs need to be split (like some sort of dependency)?

so from a bad experience with gradle it is always to add stuff gradually to avoid mistakes with these scripts , because some updates to apart can indirectly affect other parts of the code

**Comment by reidbaker** [comment]
@jesswrd also I asked @AbdeMohlbi to keep the prs small so we could quickly review (and revert if we needed to). So far I have been happy with the style.

**Review by jesswrd — APPROVED** [review]
LGTM

**Pull request merged**